### DISCUSSION TITLE
Debugging LSP

### DISCUSSION THREAD

**Discussion by mssalvatore** [LazyVim/LazyVim#2617] [Q&A] [answered]
I'm using neovim v0.9.5 with LazyVim and I'm having an issue where if I open multiple files with `nvim [-O|-o|-P]`, only the first buffer is attached to the language server. I've set the lsp log level to DEBUG and checked the `lsp.log` file, but there's no mention of any file other than the first one provided. Running `LspInfo` on shows:
```
 Language client log: /home/user/.local/state/lazy2.nvim/lsp.log
 Detected filetype:   python
 
 0 client(s) attached to this buffer: 
 
 1 active client(s) not attached to this buffer: 
 
 Client: pylsp (id: 1, bufnr: [1])
 	filetypes:       python
 	autostart:       true
 	root directory:  /home/user/test
 	cmd:             /home/user/.local/share/lazy2.nvim/mason/bin/pylsp
 
 Configured servers list: pylsp, lua_ls
```
I've tried with simpler setups to reproduce the issue, but I've not yet been able to reproduce it and pin down exactly what in my config is causing the issue. I'm not sure what steps to take next to debug. Is there another log I could check for some indication of why only the first buffer is attached to the language server? Some other command I should run for more info?

Thanks in advance!

**Comment by dpetka2001** [comment, marked as answer]
Try the following
```lua
-- ~/.config/nvim/lua/plugins/lsp.lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = function()
      vim.schedule(function() vim.cmd("LspStart") end)
    end,
  },
}
```
and test if it's the behavior you want.

**Reply by mssalvatore** [reply]
That seems to solve it! But it raises another question for me, mostly due to my lack of understanding. In `lua/plugins/python-lsp-server.lua`, I have a different config:

```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        pylsp = {
          settings = {
            pylsp = {
              configurationSources = { "flake8" },
              plugins = {
                ...
              },
            },
          },
        },
      },
    },
  },
}
```

Both configs seem to be honored. That is, it doesn't seem that one is overwriting the other. Is that because `lsp.lua` is alphabetically before `python-lsp-server.lua`? Does the `=` operator update instead of overwrite?

Is it possible to combine these two configs into one?

**Reply by dpetka2001** [reply]
Both are honored, because `lazy.nvim` (the package manager being used by LazyVim) merges the `opts` of a specific spec (in this case `nvim-lspconfig`) into a final spec even if they are in different files. 

You could also combine the 2 configs into one by probably doing
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = function(_, opts)
      opts.servers = {
        pylsp = {
          settings = {
            pylsp = {
              configurationSources = { "flake8" },
              plugins = {
                ...
              },
            },
          },
        },
      }
      vim.schedule(function() vim.cmd("LspStart") end)
    end,
  },
}
```
You should test this out as I'm not on my computer now and can't verify it.

**Reply by mssalvatore** [reply]
> Both are honored, because lazy.nvim (the package manager being used by LazyVim) merges the opts of a specific spec (in this case nvim-lspconfig) into a final spec even if they are in different files.

Got it. Thank you so much!
